nix/keys.nix

```diff
@@ -6,13 +6,34 @@ let
   keyOptionsType = types.submodule {
     options.text = mkOption {
       example = "super secret stuff";
-      type = types.str;
+      default = null;
+      type = types.nullOr types.str;
       description = ''
-        The text the key should contain. So if the key name is
-        <replaceable>password</replaceable> and <literal>foobar</literal>
-        is set here, the contents of the file
+        When non-null, this designates the text that the key should
+        contain. So if the key name is <replaceable>password</replaceable>
+        and <literal>foobar</literal> is set here, the contents of the file
         <filename>/run/keys/<replaceable>password</replaceable></filename>
         will be <literal>foobar</literal>.
+
+        NOTE: Either <literal>text</literal> or <literal>filename</literal> have
+        to be set.
+      '';
+    };
+
+    options.filename = mkOption {
+      example = "./super-secret.key";
+      default = null;
+      type = types.nullOr types.str;
+      description = ''
+        When non-null, contents of the specified file will be deployed to
+        the specified key on the target machine.  If the key name is
+        <replaceable>password</replaceable> and <literal>/foo/bar</literal> is set
+        here, the contents of the file
+        <filename>/run/keys/<replaceable>password</replaceable></filename>
+        deployed will be the same as local file <literal>/foo/bar</literal>.
```

Comment:
(nitpick): explain how this option mitigates https://github.com/NixOS/nix/issues/1307